### PR TITLE
docs/data-table

### PR DESCRIPTION
**Pull request by nekosaur**
## Description
Updated data-table and data-iterator docs

## Markup:

<details>

```vue
// Paste your FULL Playground.vue here
```
</details>

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Improvement/refactoring (non-breaking change that doesn't add any feature but make things better)

## Checklist:

- [ ] The PR title is no longer than 64 characters.
- [ ] The PR is submitted to the correct branch (`master` for bug fixes and documentation updates, `dev` for new features and breaking changes).
- [ ] My code follows the code style of this project.
- [ ] I've added relevant changes to the documentation (applies to new features and breaking changes in core library)

### CODE REVIEW

File: packages/docs/src/examples/data-iterators/usage.vue [hunk 1]
```diff
@@ -0,0 +1,168 @@
+<template>
+  <v-container fluid grid-list-md>
+    <v-data-iterator
+      :items="items"
+      :items-per-page.sync="itemsPerPage"
+      :footer-props="{ itemsPerPageOptions }"
+    >
+      <template v-slot:default="props">
```
Comment (sh7dm) [inline packages/docs/src/examples/data-iterators/usage.vue]: @nekosaur Maybe use `#` shorthand for `v-slot`?

File: packages/docs/src/examples/data-tables/customizeFooter.vue [hunk 2]
```diff
@@ -112,9 +116,3 @@
     }
   }
 </script>
-
```
Comment (sh7dm) [inline packages/docs/src/examples/data-tables/customizeFooter.vue]: Doesn't it use `mdi`?

File: packages/docs/src/examples/data-tables/customizeHeaders.vue [hunk 3]
```diff
@@ -1,49 +1,18 @@
...
-    show-select
-    item-key="name"
-    hide-default-header
     class="elevation-1"
   >
-    <template v-slot:header="{ props, on }">
-      <thead>
-        <tr>
-          <th>
-            <!-- <v-checkbox
```
Comment (sh7dm) [inline packages/docs/src/examples/data-tables/customizeHeaders.vue]: For what is this comment needed?

File: packages/docs/src/lang/en/components/DataIterators.json [hunk 4]
```diff
@@ -0,0 +1,73 @@
+{
+  "header": "# Data iterators",
+  "headerText": "The `v-data-iterator` component is used for displaying data, and shares a majority of its functionality with the `v-data-table` component. Features include sorting, searching, pagination, and selection.",
+  "examples": {
+    "usage": {
+      "desc": "The `v-data-iterator` allows you to customize exactly how to display your data. In this example we are using a grid list with cards."
+    },
+    "slots": {
+      "header": "### Slots",
+      "desc": "The `v-data-iterator` has both a header and footer slot for adding extra content"
```
Comment (sh7dm) [inline packages/docs/src/lang/en/components/DataIterators.json]:
```suggestion
      "desc": "The `v-data-iterator` component has both a header and footer slots for adding extra content to it"
```

File: packages/vuetify/dev/index.js [hunk 5]
```diff
@@ -14,6 +14,9 @@ const vm = new Vue({
   data: () => ({ isLoaded: document.readyState === 'complete' }),
   vuetify,
   router,
+  mounted () {
+    console.log(this)
```
Comment (sh7dm) [inline packages/vuetify/dev/index.js]: Why is this required?

File: packages/vuetify/src/components/VDataIterator/VDataFooter.ts [hunk 6]
```diff
@@ -1,14 +1,14 @@
...
-import { PropValidator } from 'vue/types/options'
-import Vue, { VNode, VNodeChildrenArrayContents } from 'vue'
+// Components
 import VSelect from '../VSelect/VSelect'
 import VIcon from '../VIcon'
 import VBtn from '../VBtn'
 
-// Styles
+// Helpers
+import Vue, { VNode, VNodeChildrenArrayContents } from 'vue'
```
Comment (sh7dm) [inline packages/vuetify/src/components/VDataIterator/VDataFooter.ts]: These are `// Types` not `// Helpers`

**Comment by sh7dm** [comment]
Also please do `jest -u` in order to update snapshots

File: packages/docs/src/examples/data-iterators/usage.vue (hunk 1 above)
Comment (nekosaur) [reply packages/docs/src/examples/data-iterators/usage.vue]: We've decided to not use the shorthand in examples

File: packages/docs/src/examples/data-tables/customizeFooter.vue (hunk 2 above)
Comment (nekosaur) [reply packages/docs/src/examples/data-tables/customizeFooter.vue]: Not anymore

File: packages/docs/src/examples/data-tables/customizeHeaders.vue (hunk 3 above)
Comment (nekosaur) [reply packages/docs/src/examples/data-tables/customizeHeaders.vue]: Not sure why you're commenting on removed code?

File: packages/vuetify/dev/index.js (hunk 5 above)
Comment (nekosaur) [reply packages/vuetify/dev/index.js]: Mistakenly commited

**Comment by johnleider** [comment]
* expand.vue
  * Not sure if this is intended or not 
image
* server.vue
  * when the progress linear comes in it jolts the content down by whatever its height is
* editdialog.vue
  * not working
